File: preparatoryassignmet/program1.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {
    // Check if at least one number is provided
    if (argc < 2) {
        printf("Please provide numbers as command line arguments.\n");
        return 1;
    }

    // Initialize max with the first number
    int max = atoi(argv[1]);

    // Loop through the remaining arguments
    for (int i = 2; i < argc; i++) {
        int num = atoi(argv[i]);
        if (num > max) {
            max = num;
        }
    }

    // Display the maximum number
    printf("Maximum number is: %d\n", max);

    return 0;
}

File: preparatoryassignmet/program1.py
#. Write a program to input n numbers on command line argument and 
#calculate maximum of them. 

import sys

if len(sys.argv) < 2:
    print("Please provide numbers as command line arguments.")
    sys.exit()

numbers = [float(arg) for arg in sys.argv[1:]]

max_value = max(numbers)

print(f"Maximum value is: {max_value}")